gradle/typescript.gradle

```diff
@@ -0,0 +1,40 @@
+apply plugin: 'com.github.node-gradle.node'
+
+node {
+    download = true
+    distBaseUrl = null
+    version = '16.15.0'
+    workDir = file("${rootProject.projectDir}/.gradle/nodejs")
+}
+
+task lint( type: NpmTask, dependsOn: npmInstall ) {
```

Comment:
lint must be part of clean task (clean depends on lint)

Reply:
I completely don't understand, how `lint` task is a dependency for `clean`. We must `lint` the code before we build/publish/bundle something. But `clean` is not necessary the task, that is run before one of those.

Reply:
linting does not do anything with code - thus, not affecting build: you can still build un-linted code.

Reply:
Initially, in our app's repos, `build` depends on `lint` to force our developers to fix any code problems before the changes are published. Rather controversial decision. I can see t does not fit the workflow of the XP. Do you want to keep `lint` as an independent task and run it before `test` on CI? No linting, unless it is explicitly run? (I just can't see how `clean` and `lint` tasks can be related).

Reply:
Ah, so you mean `check`, not `clean`! That makes sense. Was a bit confused, figuring out why do we need to run `lint` before `clean`.

Reply:
sorry. `check`, not `clean`.